client/fs_endpoint.go

```diff
@@ -752,12 +756,13 @@ OUTER:
 				return nil
 			case <-ctx.Done():
 				return nil
-			case err, ok := <-eofCancelCh:
+			case _, ok := <-eofCancelCh:
 				if !ok {
 					return nil
 				}
 
-				return err
+				cancelReceived = true
+				continue OUTER
```

Comment:
```suggestion
				// try to read one more frame to avoid dropped entries
				// during log rotation
				cancelReceived = true
				continue OUTER
```